hazelcast/src/main/java/com/hazelcast/spi/impl/servicemanager/impl/ServiceManagerImpl.java

```diff
@@ -148,7 +148,6 @@ private void registerDefaultServices(ServicesConfig servicesConfig) {
 
         logger.finest("Registering default services...");
         registerService(MapService.SERVICE_NAME, createService(MapService.class));
-        registerService(LockService.SERVICE_NAME, new LockServiceImpl(nodeEngine));
```

Comment:
The `MapService` relies on the `LockService`. I know that the first usage is in the `init()` method, not in the service constructor, but this still looks so wrong and racy.